Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx

```diff
@@ -1088,6 +1097,26 @@ void vtkMRMLSequenceBrowserNode::ProcessMRMLEvents(vtkObject* caller, unsigned l
     }
   else if (vtkMRMLSequenceNode::SafeDownCast(modifiedNode))
     {
+    if (this->SelectedItemNumber < 0)
+      {
+      // There was no valid selected item number. If the master sequence node now contains an item
+      // then automatically select it.
+      // This is consistent with automatically setting selected item number to INVALID_ITEM_NUMBER when all items are removed.
```

Comment:
Looking at `RemoveAllSequenceNodes`, I can see that `INVALID_ITEM_NUMBER` is set. But when I look at `RemoveSynchronizedSequenceNode`, I do not see when/where `SetSelectedItemNumber` is called with `INVALID_ITEM_NUMBER`

Reply:
It is implemented in `vtkMRMLSequenceBrowserNode::ProcessMRMLEvents`. When the master sequence node is changed then `SelectedItemNumber` may be updated.